### PR TITLE
reduce recording glitches with native sample rate

### PR DESCRIPTION
**Pull request by alexis-kochetov**

### CODE REVIEW

File: src/dotnet/Streaming.UI.Blazor/Components/AudioRecorder/constants.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const SAMPLES_PER_MS = 16;
+export const SAMPLES_PER_WINDOW_32 = SAMPLES_PER_MS * 32;
+export const SAMPLES_PER_WINDOW_30 = SAMPLES_PER_MS * 30;
+export const SAMPLE_RATE = 16000;
```
Comment (crui3er) [inline src/dotnet/Streaming.UI.Blazor/Components/AudioRecorder/constants.ts:4]: Why did sample rate become lower? It was 48000 Hz, wasn't it?

File: src/dotnet/Streaming.UI.Blazor/Components/AudioRecorder/workers/audio-vad-worker.ts [hunk 2]
```diff
@@ -147,19 +131,15 @@ async function processQueue(): Promise<void> {
...
-                const monoPcm = new Float32Array(resampled.buffer, 0, 512);
+                const monoPcm = new Float32Array(buffer, 0, SAMPLES_PER_WINDOW_32);
                 vadEvent = await nnVoiceDetector!.appendChunk(monoPcm);
             }
             else if (buffer.byteLength == SAMPLES_PER_WINDOW_30 * 4) {
-                // 30 ms at 48000 Hz
-                const monoPcm = new Float32Array(buffer, 0, 1440);
+                const monoPcm = new Float32Array(buffer, 0, SAMPLES_PER_WINDOW_30);
                 vadEvent = await webrtcVoiceDetector.appendChunk(monoPcm);
             }
```
Comment (crui3er) [inline src/dotnet/Streaming.UI.Blazor/Components/AudioRecorder/workers/audio-vad-worker.ts:144]: What will happen if none of above conditions are met?

**Review by crui3er — APPROVED** [review]

File: src/dotnet/Streaming.UI.Blazor/Components/AudioRecorder/constants.ts (hunk 1 above)
Comment (alexis-kochetov) [reply src/dotnet/Streaming.UI.Blazor/Components/AudioRecorder/constants.ts:4]: Microphone recording sample rate is 16K, but playback is 48K for Android and iOS by default, so when there is only one AudioContext mnicrophone stream has to be resampled, Now we use two different AudioContexts

File: src/dotnet/Streaming.UI.Blazor/Components/AudioRecorder/workers/audio-vad-worker.ts (hunk 2 above)
Comment (alexis-kochetov) [reply src/dotnet/Streaming.UI.Blazor/Components/AudioRecorder/workers/audio-vad-worker.ts:144]: we will have to skip VAD processing for wrong sample buffers

**Pull request merged**